src/main/java/org/beanpod/switchboard/config/DataSourceConfig.java

```diff
@@ -0,0 +1,56 @@
+package org.beanpod.switchboard.config;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import javax.sql.DataSource;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+@Configuration
+public class DataSourceConfig {
+
+  @Bean(name = "h2DataSource")
+  @Profile("test")
+  public DataSource h2DataSource() {
+    DataSourceBuilder<?> dataSourceBuilder = DataSourceBuilder.create();
+    dataSourceBuilder.driverClassName("org.h2.Driver");
+    dataSourceBuilder.url("jdbc:h2:mem:test");
+    dataSourceBuilder.username("sa");
+    dataSourceBuilder.password("");
+
+    return dataSourceBuilder.build();
+  }
+
+  @Bean(name = "mySqlDevDataSource")
+  @Profile("development")
+  public DataSource mySqlDevDataSource() {
+    DataSourceBuilder<?> dataSourceBuilder = DataSourceBuilder.create();
+    // TODO if applicable, edit credentials (CAUTION: do not push changes to version control)
+    dataSourceBuilder.url(
+        "jdbc:mysql://localhost:3306/"
+            + "switchboard?allowPublicKeyRetrieval=true&useSSL=false&serverTimezone=UTC");
+    dataSourceBuilder.username("root");
+    dataSourceBuilder.password("root");
```

Comment:
Do you know if these values can still be overridden with command line arguments?  (I can check but i don't have time right now)

Reply:
Yes we may, but I have not seen usage of args in our code. Did you program the use of cli arguments locally on your end? If so, what arg names did you use or would want to use?

Reply:
Oh, my guess is that you were prefixing your cli args with '-' which adds the property to the Spring Environment effectively overriding application.properties. I'll push an adaptation in a minute to mock the same behaviour programmatically to not impede your flow

Reply:
Done.